test/DynamoCoreTests/libGPreloaderTests.cs

```diff
@@ -65,9 +110,64 @@ public void GetInstalledASMVersions2_FindsVersionedLibGFolders()
             Assert.AreEqual("revit2020_InstallLocation", foundPath);
             //cleanup
             libG22440path.Delete(true);
-            libG22401path.Delete(true);
+            libG22500path.Delete(true);
         }
 
+        [Test]
+        public void GetInstalledASMVersions2_ASMVersionFallback()
+        {
+            var versions = new List<Version>()
+            {
+                    new Version(224,4,0),
+                    new Version(224,0,1),
+                    new Version(223,0,1),
+                    new Version(222,0,0),
+                    new Version(221,0,0),
+                    // Notice the lookup version here is different than the actual found version below
+                    // because there is no local mocked product with that specific ASM version installed
+                    new Version(225,4,0)
+            };
+
+            versions.Sort();
+            versions.Reverse();
+
+
+            var mockedInstalledASMs = new Dictionary<string, Tuple<int, int, int, int>>()
+            {
+
+                {"revit2017_InstallLocation" ,Tuple.Create<int,int,int,int>(222,0,0,0)},
+                {"revit2018_InstallLocation" ,Tuple.Create<int,int,int,int>(223,0,1,0)},
+                {"revit2019_InstallLocation" ,Tuple.Create<int,int,int,int>(224,0,1,0)},
+                {"revit2019.2_InstallLocation" ,Tuple.Create<int,int,int,int>(224,4,0,0)},
+                {"revit2020_InstallLocation" ,Tuple.Create<int,int,int,int>(225,0,0,0)},
+
+            };
+
+            var newestASM = new Version(225, 4, 0);
+            //first version should be 225.4.0
+            Console.WriteLine(versions);
+            Assert.AreEqual(newestASM, versions.First());
+
+            //mock a folder with libASMLibVersionToVersionG folders with correct names
+            var foundPath = "";
+            var rootFolder = Path.Combine(Path.GetTempPath(), "LibGTest");
+            //both versions of libG exist
+            var libG22440path = System.IO.Directory.CreateDirectory(Path.Combine(rootFolder, "LibG_224_4_0"));
+            var libG22500path = System.IO.Directory.CreateDirectory(Path.Combine(rootFolder, "LibG_225_0_0"));
+
+
+            var foundVersion = DynamoShapeManager.Utilities.GetInstalledAsmVersion2(
+                versions, ref foundPath, rootFolder, (path) => { return mockedInstalledASMs; });
+
+            // The found version in this case is a fallback of lowest version within same major which should be 225.0.0
+            Assert.AreNotEqual(newestASM, foundVersion);
```

Comment:
@alfarok See here, the `newest` (which is target version) and `found` are meant to be different.